README.md

```diff
@@ -1395,6 +1395,98 @@ This library partially supports [inline schema definition dereferencing]( http:/
 }
 ```
 
+The library partially supports `$id` for any object placed within the `definitions` property. A property within the form can refer to a definition using a `$id` field like this:
+
+```json
+{
+  "title": "A registration form",
+    "description": "A simple form example.",
+    "type": "object",
+    "required": [
+      "firstName",
+    "lastName"
+    ],
+    "definitions": {
+      "firstName": {
+        "$id": "#firstName",
+        "type": "string",
+        "title": "First name"
+      },
+      "lastName": {
+        "$id": "#lastName",
+        "type": "string",
+        "title": "Last name"
+      },
+      "age": {
+        "$id": "#age",
+        "type": "integer",
+        "title": "Age"
+      },
+      "bio": {
+        "$id": "#bio",
+        "type": "string",
+        "title": "Bio"
+      },
+      "password": {
+        "$id": "#password",
+        "type": "string",
+        "title": "Password",
+        "minLength": 3
+      }
+    },
+    "properties": {
+      "firstName": {
+        "$ref": "#firstName"
+      },
+      "lastName": {
+        "$ref": "#lastName"
+      },
+      "age": {
+        "$ref": "#age"
+      },
+      "bio": {
+        "$ref": "#bio"
+      },
+      "password": {
+        "$ref": "#password"
+      }
+    }
+}
+```
+
+Sometimes normalizing a deep nested schema can result in a subschema which is accessed by a reference and has references of its own. Such references will be followed unless the subschema refers to itself or has a circular reference between two items, then the form will throw an Error due to the malformed schema.
```

Comment:
I don't really understand what `normalizing` is supposed to mean in this context. How about something like `Subschemas can reference other subschemas, which can reference other subschemas and so on. This chain of references will be followed, but if it appears infinite or circular, an error will be thrown.`